cmd/api/src/api/v2/file_uploads.go

```diff
@@ -128,12 +128,12 @@ func (s Resources) ProcessFileUpload(response http.ResponseWriter, request *http
 	} else if fileUploadJob, err := fileupload.GetFileUploadJobByID(s.DB, int64(fileUploadJobID)); err != nil {
 		api.HandleDatabaseError(request, response, err)
 	} else if fileName, err := fileupload.SaveIngestFile(s.Config.TempDirectory(), request.Body); err != nil {
-		if errors.Is(err, fileupload.ErrInvalidIngestFileType) {
-			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusBadRequest, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
+		if errors.Is(err, fileupload.ErrInvalidJSON) {
+			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusHTTPVersionNotSupported, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
 		} else {
 			api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
 		}
-	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestContext.RequestID, int64(fileUploadJobID)); err != nil {
+	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestId, int64(fileUploadJobID)); err != nil {
```

Comment:
I know this is from before your changes, but we could make this flow a little nicer by doing:

```suggestion:
} else if fileName, err := fileupload.SaveIngestFile(s.Config.TempDirectory(), request.Body); if errors.Is(err, fileupload.ErrInvalidJSON) {
		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusHTTPVersionNotSupported, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
	} else if err != nil{
		api.WriteErrorResponse(request.Context(), api.BuildErrorResponse(http.StatusInternalServerError, fmt.Sprintf("Error saving ingest file: %v", err), request), response)
	} else if _, err = ingest.CreateIngestTask(s.DB, fileName, requestId, int64(fileUploadJobID)); err != nil {
```